packages/dd-trace/src/appsec/iast/taint-tracking/plugin.js

```diff
@@ -184,6 +205,32 @@ class TaintTrackingPlugin extends SourceIastPlugin {
     this.taintHeaders(req.headers, iastContext)
     this.taintUrl(req, iastContext)
   }
+
+  _taintDatabaseResult (result, dbOrigin, iastContext = getIastContext(storage.getStore()), name) {
+    if (!iastContext) return result
+
+    if (this._rowsToTaint === 0) return
```

Comment:
this should return `result`, no?

returning `undefined` would set `result[i] = undefined` or `result[key] = undefined`

Reply:
yeah it should. It doesn't really matter, because if `this._rowsToTaint` is 0, is not going to be into the recursive method, but I'll change it.